test/e2e/scrapeconfig_test.go

```diff
@@ -226,3 +238,88 @@ func testPromOperatorStartsWithoutScrapeConfigCRD(t *testing.T) {
 	_, err = framework.CreateOrUpdatePrometheusOperator(context.Background(), ns, []string{ns}, nil, []string{ns}, nil, false, true, true)
 	require.NoError(t, err)
 }
+
+// testScrapeConfigKubernetesNodeRole tests whether Kubernetes node monitoring works as expected
+func testScrapeConfigKubernetesNodeRole(t *testing.T) {
+	skipPrometheusTests(t)
+
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
+
+	_, err := framework.CreateOrUpdatePrometheusOperator(context.Background(), ns, []string{ns}, nil, []string{ns}, nil, false, true, true)
+	require.NoError(t, err)
+
+	p := framework.MakeBasicPrometheus(ns, "prom", "group", 1)
+	p.Spec.ScrapeConfigSelector = &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"role": "scrapeconfig",
+		},
+	}
+	_, err = framework.CreatePrometheusAndWaitUntilReady(context.Background(), ns, p)
+	require.NoError(t, err)
+
+	// For authorization
+	createServiceAccountSecret(t, "prometheus", ns)
+	secretName := "scraping-tls"
+	// apply authorized certificate and key to k8s as a Secret
+	createSecret(t, secretName, ns)
+
+	sc := framework.MakeBasicScrapeConfig(ns, "scrape-config")
+	sc.Spec.Scheme = pointer.String("HTTPS")
+	sc.Spec.Authorization = &monitoringv1.SafeAuthorization{
+		Credentials: &v1.SecretKeySelector{
+			LocalObjectReference: v1.LocalObjectReference{
+				Name: "prometheus-sa-secret",
+			},
+			Key: "token",
+		},
+	}
+	sc.Spec.TLSConfig = &monitoringv1.SafeTLSConfig{
+		InsecureSkipVerify: true,
```

Comment:
On kind cluster I tried, 

```Get "https://172.18.0.8:10250/metrics": tls: failed to verify certificate: x509: cannot validate certificate for 172.18.0.8 because it doesn't contain any IP SANs```

Reply:
We discussed offline with @slashpai but in this case, it's ok to discard TLS verification.